### PR TITLE
New features and serious ES6 refactoring

### PR DESCRIPTION
**Pull request by redpfire**
Hey, I wanted to use your bot in my server so I looked at the code and improved it a lot.

Let me know what you think!

Changelog:
- New extended config featuring a customizable log channel, custom action aft er a violation, and more
- Full ES6 module architecture (early; needs improving for sure)
- Bot will send an Embed to the custom log channel or channel the message originated from
- Three choose-able actions after a violation: **DELETE, BAN or IGNORE**
- Some new filters originating from my case studies
- License fix in `package.json`
- A lot of refactoring :)

### CODE REVIEW

File: filter.json [hunk 1]
```diff
@@ -25,5 +25,12 @@
   "treadeoffer",
   "discordgifts",
   "discorcl",
-  "dlscord.\n"
+  "dlscord.\n",
+  "Discord Nitro",
+  "Nitro",
+  "bit.ly",
+  "for Free",
```
Comment (redpfire) [inline filter.json:32]: Note: these should be case insensitive but that's a further improvement

Comment (AerGameChannel) [reply filter.json:32]: This can cause false triggering. For example, members can talk about discord nitro and get banned for "scam".

**Comment by AerGameChannel** [comment]
Okay, thank you very much for contributing to my project. I really appreciate it.

File: filter.json (hunk 1 above)
Comment (redpfire) [reply filter.json:32]: Okay, that's valid. In our server we don't talk in English so it wouldn't false trigger.

**Pull request merged**